Builds/CMake/deps/Protobuf.cmake

```diff
@@ -41,6 +41,7 @@ if (local_protobuf OR NOT Protobuf_FOUND)
       -Dprotobuf_DEBUG_POSTFIX=_d
       -Dprotobuf_WITH_ZLIB=$<IF:$<BOOL:${has_zlib}>,ON,OFF>
       $<$<BOOL:${CMAKE_VERBOSE_MAKEFILE}>:-DCMAKE_VERBOSE_MAKEFILE=ON>
+      $<$<BOOL:${unity}>:-DCMAKE_UNITY_BUILD=ON}>
```

Comment:
I don't feel like ABI-agnostic build options specific to one package should affect build options of other packages, but I'm not going to contest strongly.

Reply:
This is our private/local/nih build of protobuf. It's certainly possible that some external project will not be ready for unity builds (and fail spectacularly), in which case we'd want to not pass the option, or force it to OFF - but so far these few seem to be fine. The goal of unity is to speed up compilation. If you asked for unity AND you doing local deps building, why wouldn't you want unity there too? As always, you can bypass the local build of most deps by pre-building/installing.